webapp/src/main/java/com/diffblue/watchtower/util/ReportParserUtil.java

```
@@ -0,0 +1,55 @@
+package com.diffblue.watchtower.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+
+import com.diffblue.watchtower.reports.interfaces.ReportParser;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ReportParserUtil {
+    private static final Logger logger = LoggerFactory.getLogger(ReportParserUtil.class);
+    private Map<String, ReportParser> parsers;
+
+    @Resource
+    PropertiesUtil propertiesUtil;
+
+    @PostConstruct
+    public void setup () {
+        parsers = Collections.synchronizedMap(new HashMap<>());
+        logger.debug("{} created", this.getClass());
+    }
+
+    private ReportParser initAndReturnParser (String reportId) {
+        logger.info("Initialising parser for report {}", reportId);
+
+        Optional<String> parserClassName = propertiesUtil.getReportParser(reportId);
+        if (parserClassName.isPresent()) {
+            try {
+                ReportParser parser = (ReportParser) Class.forName(parserClassName.get()).newInstance();
+                parsers.put(reportId, parser);
+                return parser;
+            } catch (Exception e) {
+                logger.error("An error occurred during the initialisation of parser for report id {}", reportId, e);
+            }
+        }
+
+        logger.error("Could not initialise parser for report id {}", reportId);
+        return null;
+    }
```

Would it not be better to throw the exception here instead of handling it and returning null?